lightning/src/chain/mod.rs

```diff
@@ -187,68 +187,81 @@ pub trait Confirm {
 	fn get_relevant_txids(&self) -> Vec<Txid>;
 }
 
-/// An error enum representing a failure to persist a channel monitor update.
+/// An enum representing the status of a channel monitor update persistence.
 #[derive(Clone, Copy, Debug, PartialEq)]
-pub enum ChannelMonitorUpdateErr {
+pub enum ChannelMonitorUpdateStatus {
+	/// The update has been durably persisted and all copies of the relevant [`ChannelMonitor`]
+	/// have been updated.
+	///
+	/// This includes performing any `fsync()` calls required to ensure the update is guaranteed to
+	/// be available on restart even if the application crashes.
+	Complete,
 	/// Used to indicate a temporary failure (eg connection to a watchtower or remote backup of
 	/// our state failed, but is expected to succeed at some point in the future).
```

Comment:
so there's a case, where we think it might succeed in future and put it InProgress.

At later point in time:
if it does succeed => good, restore the channel
if it fails => doc doesn't mention what to do.